runtime/src/main/java/org/corfudb/runtime/collections/ManagedTxnContext.java

```diff
@@ -675,16 +683,18 @@ void deleteRecord(@Nonnull String tableName,
                 throw new IllegalArgumentException("Non-null payload sent for delete validation");
             }
 
-            final Message.Builder builder = deltaMetadata.toBuilder();
             for (Descriptors.FieldDescriptor fieldDescriptor : deltaMetadata.getDescriptorForType().getFields()) {
                 if ("revision".equals(fieldDescriptor.getName())) {
                     if (oldRecord == null || oldRecord.getMetadata() == null) {
                         return null;
                     } else {
-                        Long prevRevision = (Long) oldRecord.getMetadata().getField(fieldDescriptor);
-                        Long givenRevision = (Long) deltaMetadata.getField(fieldDescriptor);
-                        if (givenRevision == 0 || // Do not validate revision if field isn't set
-                                (givenRevision > 0 && // Validate revision only if set
+                        Long prevRevision = ((Int64Value) oldRecord.getMetadata().getField(fieldDescriptor)).getValue();
+                        Long givenRevision = null;
+                        if (deltaMetadata.hasField(fieldDescriptor)) {
+                            givenRevision = ((Int64Value) deltaMetadata.getField(fieldDescriptor)).getValue();
+                        }
+                        if (givenRevision == null || // Do not validate revision if field isn't set
+                                (givenRevision >= 0 && // Validate revision only if set
```

Comment:
is ">=0" still needed here?

Reply:
I was thinking about ignoring negative numbers, but you're actually right its not needed. Removed!